src/vs/code/browser/workbench/workbench.ts

```diff
@@ -401,11 +401,15 @@ class WindowIndicator implements IWindowIndicator {
 (function () {
 	const route = parseGitHubUrl(window.location.href);
 	const config: IWorkbenchConstructionOptions & { folderUri?: UriComponents, workspaceUri?: UriComponents } = {
-		folderUri: URI.from({ scheme: "github1s", path: '/', authority: `${route.owner}+${route.repo}+${route.branch}` }),
+		/**
+		 * We couldn't figure out the correct branch here since branch might has `/` i.e. `feat/new-feat-1`.
+		 * So we have to use the `route.path` in authority and figure out branch in github1s extension.
+		 */
+		folderUri: URI.from({ scheme: "github1s", path: '/', authority: `${route.owner}+${route.repo}+${route.path}` }),
```

Comment:
@conwnet could you please confirm whether this assumption is correct?

The code would be significantly simpler if we could call `commandService.executeCommand('github1s.get-github-branches', window.location.href)` here.

Reply:
yes, but we can't call `commandService` here, I want to remove the `branch` part from URI authority, and deal with `branch` in github1s extension, let me have a try

Reply:
I did try this direction. The only missing part is that we use `path = ‘/‘` to indicate the repo’s root which will lose the original pathname. A possible solution is using query or fragment in URI to indicate the information.

Reply:
yes, after reading the documentation of [VS Code Extension Source Control](https://code.visualstudio.com/api/extension-guides/scm-provider), I realized the importantce for **Exact URI** of resouce. I adjusted that

Reply:
change the `authority` as format `{owner}/{repo}/{ref}`, we assume the empty `authority` means github1s should get it from `window.location.href` https://github.com/xcv58/github1s/pull/6